Mathlib/SetTheory/Cardinal/Ordinal.lean

```diff
@@ -1445,3 +1447,80 @@ theorem extend_function_of_lt {α β : Type _} {s : Set α} (f : s ↪ β) (hs :
 -- end Bit
 
 end Cardinal
+
+section Initial
+
+namespace Ordinal
+
+open Cardinal
+
+/--
+`initial i` (notation: `ω_ i`) gives the first ordinal of cardinality
+`aleph i`. Thus `ω_ 0 = ω` and `ω_ 1` is the first uncountable ordinal.
+You can also write `ω_ 1` as `ω₁`.
+-/
+def initial (x : Ordinal.{u}) : Ordinal.{u} := (aleph x).ord
+
+@[inherit_doc]
+scoped notation "ω_" => Ordinal.initial
+@[inherit_doc]
+scoped notation "ω₁" => Ordinal.initial 1
+
+@[simp]
+theorem card_initial (i : Ordinal): card (initial i) = aleph i := card_ord _
+
+/--
+The first (infinite) initial ordinal is `ω`.
+-/
+@[simp]
+lemma initial0 : ω_ 0 = ω := by
+  unfold initial
+  simp
+
+/--
+An initial ordinal is a limit ordinal.
+-/
+lemma isLimit_initial (i : Ordinal): (ω_ i).IsLimit := ord_isLimit (aleph0_le_aleph i)
+
+lemma omega_lt_omega1 : ω < ω₁ := by
+  have := (ord_lt_ord.mpr (aleph0_lt_aleph_one))
+  rw [ord_aleph0] at this
+  exact this
+
+end Ordinal
+
+end Initial
+
+section OrdinalIndices
+/-!
+### Cardinal operations with ordinal indices
+
+Results on cardinality of ordinal-indexed families of sets.
+-/
+namespace Cardinal
+
+open scoped Cardinal
+
+/--
+Bounding the cardinal of an ordinal-indexed union of sets.
+-/
+lemma mk_iUnion_Ordinal_le_of_le {β : Type _} {κ : Cardinal} {i : Ordinal}
+  (hi : i ≤ κ.ord) (hκ : ℵ₀ ≤ κ) (A : Ordinal → Set β)
+  (hA : ∀ j < i, #(A j) ≤ κ) :
+  #(⋃ j < i, A j) ≤ κ := by
+  have : (⋃ j < i, A j) = ⋃ j : i.out.α, A (@typein _ (· < ·) (by infer_instance) j)
+  · ext x; constructor
+    · rintro ⟨_, ⟨j, rfl⟩, _, ⟨hji, rfl⟩, hx⟩
+      refine mem_iUnion.2 ⟨enum (· < ·) j ?_, ?_⟩
+      · rwa [Ordinal.type_lt]
+      simpa using hx
+    · rintro ⟨_, ⟨j, rfl⟩, hx⟩
+      exact mem_iUnion₂.2 ⟨_, typein_lt_self j, hx⟩
+  rw [this]
```

Comment:
mathlib4 style + golf
```suggestion
    (hi : i ≤ κ.ord) (hκ : ℵ₀ ≤ κ) (A : Ordinal → Set β)
    (hA : ∀ j < i, #(A j) ≤ κ) :
    #(⋃ j < i, A j) ≤ κ := by
  simp_rw [← mem_Iio, biUnion_eq_iUnion, iUnion, iSup, ← i.enumIsoOut.symm.surjective.range_comp]
```
I would suggest also use the names `c` and `o` here. It's natural to use `i : Ordinal` above where it serves as an index, but not so much here.
Also suggest changing `hi : i ≤ κ.ord` to `hi : i.card ≤ κ` to makes it more general.